errorDisplay.js

```diff
@@ -3,108 +3,87 @@
-		widgetsErrors.forEach(function (lineWidget) {
-			if (lineWidget) {
-				lineWidget.clear();
-			}
-		});
-		widgetsErrors = [];
+		var editor = EditorManager.getActiveEditor();
+		editor._codeMirror.getDoc().removeLineClass(line, "background", "errorHighlight");
 	}
 
 	//Function that adds a button on the gutter (on given line nubmer) next to the line numbers
-	function showGutter(lineStart){
-		if(gutters.length === 0){
-			var editor = EditorManager.getFocusedEditor();
-			var $errorDiv = $("<div class='error'/>");
-			var $errorMarker = $("<span class='errorButton'/>");
-			var foundGutters = ["errorButton"];
+	function showButton(line){
+		var editor = EditorManager.getActiveEditor();
+		var errorMarker = document.createElement("div");
+		errorMarker.className = "errorButton errorText";
+		errorMarker.innerHTML = "!";
 
-			$errorDiv.append($errorMarker);
-			$errorMarker.addClass("errorText");
-			$errorMarker.text("!");
+		editor._codeMirror.setGutterMarker(line, "errors", errorMarker);
 
-			gutters.push(editor._codeMirror.setGutterMarker(lineStart, "errorButton", $errorDiv[0]));
+		//Show tooltips message
+		$(".errors").tooltipsy({content : "Click error icon for details", alignTo: "cursor", offset: [10, -10]});
+	}
 
-			editor._codeMirror.setOption("gutters", foundGutters);
-			//Show tooltips message
-			$(".CodeMirror-linenumbers").tooltipsy({content : "Click button for information"}); 
+	// Function that removes gutter button
+	function removeButton(){
+		var editor = EditorManager.getActiveEditor();
+		editor._codeMirror.clearGutter("errors");
 
-			$(".CodeMirror-gutter").addClass("gutterCursor");
+		//Destroy tooltips instance
+		var tooltips = $(".errors").data("tooltipsy");
+		if(tooltips) {
+			tooltips.destroy();
 		}
 	}
 
-	//Function that removes gutter button
-	function removeGutter(){
-		var editor = EditorManager.getFocusedEditor();
-		gutters = [];
-		editor._codeMirror.clearGutter("errorButton");
-		//Destroy tooltips instance
-		$(".CodeMirror-linenumbers").data("tooltipsy").destroy();
-		//Changes cursor back to default
-		$(".CodeMirror-gutter").removeClass("gutterCursor");
+	function showDescription(error) {
+		var editor = EditorManager.getActiveEditor();
+		var description = document.createElement("div");
+		description.className = "errorPanel";
+		description.innerHTML = Mustache.render(lineWidgetHTML, {"error": error.message});
+		var options = {coverGutter: false, noHScroll: false, above: false, showIfHidden: false};
+
+		currentErrorWidget = editor._codeMirror.addLineWidget(error.line, description, options);
 	}
-	
-	exports.markErrors = markErrors;
-	exports.clearErrors = clearErrors;
-	exports.showWidget = showWidget;
-	exports.showGutter = showGutter;
-	exports.removeGutter = removeGutter;
-	exports.removeWidget = removeWidget;
-});
\ No newline at end of file
+
+	function hideDescription() {
+		if(currentErrorWidget) {
+			currentErrorWidget.clear();
+			currentErrorWidget = undefined;
```

Comment:
Use `null` for your rvalue unless you actually want `undefined` (which you don't).  `undefined` means that it's never been defined, not that you've defined it as `undefined`.